### PR TITLE
feat(ShowMore): Add callback function for panel toggle

### PR DESCRIPTION
**Pull request by hklo**
Add in a callback function when panel been toggle.

### CODE REVIEW

**Comment by patrickliu-sa** [comment]
Will this one replace `onPanelOpen` evantually? 🤔

**Comment by hklo** [comment]
Yup, thinking to replace it, or I can replace it now?

**Comment by patrickliu-sa** [comment]
@ChanJit @ZhenYongSeekAsia @chiwongseekasia I can thinking of job detail page using this component so far. Before we take out `onPanelOpen` from style guide is there anything using this callback e.g. "tracking on open rate" kind of thing in discovery?

**Comment by hklo** [comment]
Added in the deprecation warning

File: react/ShowMore/ShowMore.js [hunk 1]
```diff
@@ -47,15 +48,21 @@ export default class ShowMore extends Component {
...
+    const { isPanelOpened } = this.state;
+    const { onPanelOpen, onPanelToggle = () => {} } = this.props;
     e.preventDefault();
 
-    if (!this.state.isPanelOpened) {
-      onPanelOpen();
+    if (!isPanelOpened) {
+      if (typeof onPanelOpen === 'function') {
+        console.warn('Property of onPanelOpen has been deprecated. Use onPanelToggle instead.');
+        onPanelOpen();
```
Comment (RayChew) [inline react/ShowMore/ShowMore.js]: after run a console.warn still continue to run the function?

Comment (hklo) [reply react/ShowMore/ShowMore.js]: Yes. Console warn put in to notified others that this function will be deprecated in the future. So they can change it respectively

**Pull request merged**